magda-web-client/src/Components/Admin/Connectors.js

```diff
@@ -1,45 +1,155 @@
+              {this.state.newConnectorFormIsOpen && this.renderConnectorForm()}
+            </div>)
+  }
+
+  openConnectorForm(){
+    this.setState({
+      newConnectorName: '',
+      newConnectorType: '',
+      newConnectorFormIsOpen: true
+    })
+  }
+
+  closeConnectorForm(){
+    this.props.resetConnectorForm();
+    this.setState({
+      newConnectorFormIsOpen: false,
+    })
+  }
+
+  renderConnectorForm(){
+    const disableSubmit = this.props.error || this.props.isFetching || !this.state.newConnectorName.length || !this.state.newConnectorType.length;
+    return  <div className='create-connector-form-wrapper' onClick={()=>{this.closeConnectorForm()}}>
+              <form className='create-connector-form' onClick={(event)=>{event.stopPropagation()}}>
+              {this.props.error && <div className="alert alert-danger">{this.props.error}</div>}
+              <div>
+                <label>
+                    Name:
+                </label>
+                    <input type="text" name="name" onBlur={(event)=>{this.validateField(event.target.value, 'name')}} onChange={(event)=>{this.setState({newConnectorName: event.target.value, error: null})}} value={this.state.newConnectorName}/>
+              </div>
+              <div>
+                <label>
+                    Type:
+                </label>
+                    <input type="text" name="type" onBlur={(event)=>{this.validateField(event.target.value, 'type')}} onChange={(event)=>{this.setState({newConnectorType: event.target.value, error: null})}} value={this.state.newConnectorType}/>
+                </div>
+                <input type="button" value="Submit" onClick={()=>this.submitNewConnector()} disabled={disableSubmit} className='btn btn-primary' />
+            </form>
+          </div>
+  }
+
+  validateField(value, type){
+    if(type === 'name'){
+      this.props.validateConnectorName(value);
+    } else if(type === 'type'){
+      this.props.validateConnectorType(value);
+    }
+  }
+
+  submitNewConnector(){
+      this.props.createNewConnector({
+        name: this.state.newConnectorName,
+        type: this.state.newConnectorType,
+        id: encodeURI(this.state.newConnectorName)
+      });
+      this.setState({
+        newConnectorFormIsOpen: false
+      })
+  }
+
+  toggleConnector(connector){
+    const action = connector.job ? 'stop' : 'start';
+    this.props.updateConnectorStatus(connector.id, action)
+  }
+
+  deleteConnector(connector){
+    this.props.deleteConnector(connector.id)
+  }
 
+  renderConnector(connector){
+    return (<tr key={connector.id} className={`${connector.name === this.state.newConnectorName ? 'success': ''}`}>
+    <td>{connector.name}</td>
+    <td>{connector.type}</td>
+    <td>{connector.sourceUrl}</td>
+    <td><button className={`btn ${!connector.job ? 'btn-success': 'btn-warning'}`} type='button' onClick={this.toggleConnector.bind(this, connector)}>{!connector.job ? 'Start' : 'Stop'}</button></td>
```

Comment:
this needs to be something like `connector.job && connector.job.status === 'active'`. Connectors can also have `job.status === 'succeeded'` and `job.status === failed` :).